src/js/component.js

```diff
@@ -186,6 +186,8 @@ class Component {
    * @return {Object}     A NEW object of this.options_ and obj merged
    */
   options(obj) {
+    log.warn('this.options() has been deprecated and will be moved to the constructor in 6.0');
```

Comment:
We might want to consider bringing in [util-deprecate](https://www.npmjs.com/package/util-deprecate) so warnings are only logged once. Not required for this PR.